### ISSUE TITLE
[spaces] Hide link sharing option if not allowed

### ISSUE THREAD

**Issue by hurradieweltgehtunter** [owncloud/web#8701]
### Steps to reproduce
1. Enter space where you are a member with role “Viewer”

### Expected behavior
According to our guidelines on owncloud.design non-available options should be hidden.
Therefore, the area “Share via Link” should be hidden for users who do not have this permission.

<img width="436" alt="grafik" src="https://user-images.githubusercontent.com/16665512/227499780-79248ec2-1eaf-49a0-b76e-ebba4b34795a.png">

### Actual behavior
The option is shown as disabled.

<img width="433" alt="grafik" src="https://user-images.githubusercontent.com/16665512/227499546-7e3f6488-03c6-48be-a671-644f205b6a8c.png">

**Comment by JammingBen** [comment]
Not quite sure here because existing links would be displayed in that section as well. Do we really want to hide the section when no links given? The same applies to regular shared resources when the share receiver doesn't have the permission to create links.

**Comment by hurradieweltgehtunter** [comment]
![grafik](https://user-images.githubusercontent.com/16665512/228477803-7e78ec1d-e242-4301-9e3b-18ddccdea7b9.png)

Ok, this is unexpected.
My point of view: Why can I copy and forward a sharing link but above it reads "I am not allowed to share this folder" ?
Isn't that contradictory

**Comment by JammingBen** [comment]
The text only refers to the permission for *creating* links, but I totally agree that this is confusing as it is. Maybe change the text to `You don't have permission to create links for this folder.`? Again, I'm not a fan of hiding this section, because it could give the impression that this resource can't have links at all. @tbsbdr Any opinions on this?

**Comment by hurradieweltgehtunter** [comment]
I wasn't aware (and didn't expect) of having sharing links if I am a viewer without sharing permission.
But since the viewer permission should grant only "view" and not "re-share" they shouldn't be visible IMO.
Let's spawn @tbsbdr :)

**Comment by kulmann** [comment]
Same opinion like @JammingBen about hiding the panel entirely: if link sharing is allowed at all (controlled via capability) the user expects to have the links panel available. If that's missing, in my experience that typically causes bug reports like "bug! bug! my link sharing panel is missing!".

Also agree that the text is misleading. We need to change the wording.

From what I know existing links should be available. "resharing" is only a feature for regular shares with users/groups. But I get that this is not 100% expected. I'd keep it like it is and only change the wording. Open for different opinions.

**Comment by tbsbdr** [comment]
yep, I'd favour a re-wording as well as long we have not reworked the sharing & links section.
Wording proposal:
"You can use existing links, but you do not have permission to create links."

**Comment by hurradieweltgehtunter** [comment]
> If that's missing, in my experience that typically causes bug reports like "bug! bug! my link sharing panel is missing!".

Wouldn't mean that we need to update our design guidelines? From https://owncloud.design/#/Interface%20Guidelines :
```
Choose hiding over disabling interactive elements

Under normal circumstances and apart from "Save" actions it is good practice to not disable elements that are not accessible to a user (for example due to a lack of role privileges on their account), but to hide irrelevant controls altogether.
```

I get mixed feelings here :)


> `"You can use existing links, but you do not have permission to create links." `any objections?

`You do not have permission to create links` would be sufficient IMO. Allowance of existing links is self-explanatory.

**Comment by kulmann** [comment]
> > If that's missing, in my experience that typically causes bug reports like "bug! bug! my link sharing panel is missing!".
> 
> If that's our perception, wouldn't mean that we need to update our design guidelines? From https://owncloud.design/#/Interface%20Guidelines :
> 
> ```
> Choose hiding over disabling interactive elements
> 
> Under normal circumstances and apart from "Save" actions it is good practice to not disable elements that are not accessible to a user (for example due to a lack of role privileges on their account), but to hide irrelevant controls altogether.
> ```

I don't know if such a design principle makes sense at all - IMO the answer to that particular detail is always "it depends".

**Comment by kulmann** [comment]
> `You do not have permission to create links` would be sufficient IMO. Allowance of existing links is self-explanatory.

Fine by me, let's do that :-)
